src/measure_theory/lp_space.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2020 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Rémy Degenne.
+-/
+import measure_theory.l1_space
+import analysis.special_functions.pow
+
+/-!
+# ℒp space
+
+This file describes properties of measurable functions with finite seminorm
+`(∫⁻ a, (nnnorm (f a))^(p : ℝ) ∂μ) ^ (1/p)` for `p:ℝ` with `1 ≤ p`.
+
+## Main definitions
+
+* `mem_ℒp f p μ` : the function `f` has finite p-seminorm for measure `μ`, for `p:ℝ` such that
+                  `hp1 : 1 ≤ p`,
+
+## Notation
+
+* `snorm f p μ` : `(∫⁻ a, (nnnorm (f a))^(p : ℝ) ∂μ) ^ (1/p)` for `f : α → β`
+
+-/
+
+open measure_theory
+
+noncomputable theory
+
+namespace ℒp_space
+
+variables {α β γ: Type*} [measurable_space α] {μ : measure α}
+  [measurable_space β] [normed_group β]
+  [normed_group γ]
+  {p : ℝ}
+
+section ℒp_space_definition
+
+/-- The property that f belongs to ℒp for measure μ and real p -/
+def mem_ℒp (f : α → β) (p : ℝ) (μ : measure α) : Prop :=
+measurable f ∧ ∫⁻ a, (nnnorm (f a)) ^ p ∂μ < ⊤
+
+/-- seminorm on ℒp -/
+def snorm (f : α → γ) (p : ℝ) (μ : measure α) : ennreal := (∫⁻ a, (nnnorm (f a))^p ∂μ) ^ (1/p)
+
+lemma lintegral_rpow_nnnorm_eq_rpow_snorm {f : α → γ} (hp0_lt : 0 < p) :
+  ∫⁻ a, (nnnorm (f a)) ^ p ∂μ = (snorm f p μ) ^ p :=
+begin
+  unfold snorm,
+  rw [←ennreal.rpow_mul, one_div, inv_mul_cancel, ennreal.rpow_one],
+  symmetry, exact ne_of_lt hp0_lt,
+end
+
+end ℒp_space_definition
+
+lemma mem_ℒp_one_iff_integrable : ∀ f : α → β, mem_ℒp f 1 μ ↔ integrable f μ :=
+begin
+  intro f,
+  unfold integrable, unfold has_finite_integral, unfold mem_ℒp,
+  simp only [ennreal.rpow_one, nnreal.coe_one],
+end
+
+section top
+
+lemma snorm_lt_top {f : α → β} (hp0 : 0 ≤ p) (hfp : mem_ℒp f p μ) : snorm f p μ < ⊤ :=
+begin
+  unfold snorm,
+  refine ennreal.rpow_lt_top_of_nonneg _ (ne_of_lt hfp.right),
+  rw [one_div, inv_nonneg],
+  exact hp0,
+end
+
+lemma snorm_ne_top {f : α → β} (hp0 : 0 ≤ p) (hfp : mem_ℒp f p μ) : snorm f p μ ≠ ⊤ :=
+ne_of_lt (snorm_lt_top hp0 hfp)
+
+lemma lintegral_rpow_nnnorm_lt_top_of_snorm_lt_top {f : α → γ} (hp0_lt : 0 < p)
+  (hfp : snorm f p μ < ⊤) :
+  ∫⁻ a, (nnnorm (f a)) ^ p ∂μ < ⊤ :=
+begin
+  rw lintegral_rpow_nnnorm_eq_rpow_snorm hp0_lt,
+  exact ennreal.rpow_lt_top_of_nonneg (le_of_lt hp0_lt) (ne_of_lt hfp),
+end
+
+lemma mem_ℒp_of_snorm_lt_top {f : α → β} (hp0_lt : 0 < p) (hfm : measurable f)
+  (hfp : snorm f p μ < ⊤) :
+  mem_ℒp f p μ :=
+begin
+  split,
+  exact hfm,
+  exact lintegral_rpow_nnnorm_lt_top_of_snorm_lt_top hp0_lt hfp,
+end
```

Comment:
```suggestion
⟨hfm, lintegral_rpow_nnnorm_lt_top_of_snorm_lt_top hp0_lt hfp⟩ 
```
is exactly the same as your proof, but more concise. (This is the term-mode version of your tactic-mode proof).